src/pages/api/users/[...id].ts

```diff
@@ -1,49 +1,44 @@
-import type { NextApiRequest, NextApiResponse } from 'next'
-import next from 'next';
+import type { NextApiRequest, NextApiResponse } from "next";
+import next from "next";
 import db from "../db";
 import * as models from "../models";
-import { User } from '@/model';
-
-export default async function userHandler(
-  req: NextApiRequest,
-  res: NextApiResponse
-) {
-  const { query, method } = req
-
-  const id: string[] = query.id as string[]
-  const name = query.name as string
-
-  switch (method) {
-    case 'GET':
-      // Get data from your database
-      const resp = await handleGet(id);
-      if (resp) {
-        res.status(200).json(resp);
-      } else {
-        res.status(404).end();
-      }
-      break
-    case 'PUT':
-      // Update or create data in your database
-      res.status(200).json({ name: name || `User ${id}` })
-      break
-    default:
-      res.setHeader('Allow', ['GET', 'PUT'])
-      res.status(405).end(`Method ${method} Not Allowed`)
-  }
-}
+import { User } from "@/model";
+import nextConnect from "next-connect";
+
+export default nextConnect()
+  .put(async (req: NextApiRequest, res: NextApiResponse) => {
+    const { query } = req;
+
+    const id: string[] = query.id as string[];
+    const name = query.name as string;
+    res.status(200).json({ name: name || `User ${id}` });
+  })
+  .get(async (req: NextApiRequest, res: NextApiResponse) => {
+    const { query } = req;
+
+    const id: string[] = query.id as string[];
+
+    const resp = await handleGet(id);
+    if (resp) {
+      res.status(200).json(resp);
+    } else {
+      res.status(404).end();
+    }
+  });
 
 export async function handleGet(routes: string[]) {
   const userId = routes[0];
   switch (routes.length) {
     case 3:
-      switch (routes[1]) {
+      switch (
```

Comment:
I think having these switches causes the api to stall. 

Can we move it to next connect, even if it means different routes altogether. We can change it in the ui too

Reply:
okay

Reply:
done